File: bench/benches/float.lua
-- mandelbrot escape counts over a 64x64 grid (see bench/benches/float.l).
package.path = (arg[0]:match("(.*/)") or "./") .. "../lib/?.lua;" .. package.path
local bench = require("bench")
local function mand(cx, cy)
  local zx, zy, it = 0.0, 0.0, 0
  while it < 100 and zx * zx + zy * zy <= 4.0 do
    local nzx = zx * zx - zy * zy + cx
    local nzy = 2.0 * zx * zy + cy
    zx = nzx; zy = nzy; it = it + 1
  end
  return it
end
bench("float", function()
  local s = 0
  for px = 0, 63 do
    for py = 0, 63 do
      s = s + mand(-2.0 + px * 0.046875, -1.5 + py * 0.046875)
    end
  end
  return s
end)

File: bench/benches/float.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "lib"))
from bench import bench

# mandelbrot escape counts over a 64x64 grid (see bench/benches/float.l).
def mand(cx, cy):
    zx = 0.0
    zy = 0.0
    it = 0
    while it < 100 and zx * zx + zy * zy <= 4.0:
        nzx = zx * zx - zy * zy + cx
        nzy = 2.0 * zx * zy + cy
        zx = nzx
        zy = nzy
        it += 1
    return it

def work():
    s = 0
    for px in range(64):
        for py in range(64):
            s += mand(-2.0 + px * 0.046875, -1.5 + py * 0.046875)
    return s

bench("float", work)
